docs/code_insights/references/incomplete_data_points.mdx

```diff
@@ -24,6 +24,50 @@ You can read more about this case in our [limitations](/code_insights/explanatio
 
 If the data is available, the error alert will inform you which times the search has timed out.
 
+## Strategies for very large repositories
+
+When dealing with large repositories, several strategies can help identify and manage search limitations effectively.
+
+The goal is to find a query that can execute successfully, and then ramp up complexity until we find the breaking point.
+
+### Use Code Search to test your query
+
+You can use Code Search to test the query against a particular timestamp in a given repository.
+
+Since Code Insights computes data points for twelve datapoints in the give time range,
```

Comment:
Thank you! I was hoping for some info like that, since I had a suspicion about older commits taking longer to search. I'll update some text above to work your suggestion in. Let me know if that sounds good :)